pkg/storage/storage.go

```diff
@@ -548,6 +559,39 @@ func (is *ImageStore) GetBlobUpload(repo string, uuid string) (int64, error) {
 	return fi.Size(), nil
 }
 
+// PutBlobChunkStreamed appends another chunk of data to the specified blob. It returns
+// the number of actual bytes to the blob.
+func (is *ImageStore) PutBlobChunkStreamed(repo string, uuid string, body io.Reader) (int64, error) {
+	if err := is.InitRepo(repo); err != nil {
+		return -1, err
+	}
+
+	blobUploadPath := is.BlobUploadPath(repo, uuid)
+
+	_, err := os.Stat(blobUploadPath)
+	if err != nil {
+		return -1, errors.ErrUploadNotFound
+	}
+
+	file, err := os.OpenFile(
+		blobUploadPath,
+		os.O_WRONLY|os.O_CREATE,
+		0600,
+	)
+	if err != nil {
+		is.log.Fatal().Err(err).Msg("failed to open file")
+	}
+	defer file.Close()
+
+	if _, err := file.Seek(0, 2); err != nil {
```

Comment:
Can we use os.SEEK_END here?

Reply:
os.SEEK* deprecated apparently, use io.Seek*

pushing an updated PR

Reply:
Oh, right, thanks!